### ISSUE TITLE
cannot configure file rotation for Filebeat/Metricbeat specific logs under agent

### ISSUE THREAD

**Issue by amitaiporat** [elastic/beats#27853]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.14

**Plugins installed**: []

**JVM version** (`java -version`):

**OS version** (`uname -a` if on a Unix-like system): Windows 10 Pro

**Description of the problem including expected versus actual behavior**:
Elastic agent on Windows produces large agent log files. Changing log level can reduce this but does not root cause the problem below.

Setting `rotateeverybytes`  in elastic-agent.yml  does not change rotate policy for all log files. Appears to work for elastic-agent* logs as well as filebeat_monitoring* metricbeat_monitoring* logs. It does not however work for filebeat-json* logs or metricbeat-json* logs in the next dir **default/**

TL;DR  It looks like the updated parameter rotateeverybytes only applies to agent specific logs (including the fb and mb monitoring logs) but not to filebeat and metricbeat logs. 

Desired is to be able to  configure file rotation / max log file size from the Filebeat/Metricbeat specific logs under agent.
My suspicion is it has something to do with the fact that the files that don't match the rotation policy are also files that are not ".log" files (filebeat-json, metricbeat-json)

**Steps to reproduce**: Deploy Elastic agent on Windows host and enable system + windows integration. 
Set rotateeverybytes value in elastic-agent.yml to 10MB (default) or any value... 

<img width="616" alt="Screen Shot 2021-09-09 at 11 58 33 AM" src="https://user-images.githubusercontent.com/2553378/132746574-888c0472-b8ea-44e4-80d5-a7d082692fc7.png">

Please include a *minimal* but *complete* recreation of the problem,
including (e.g.) index creation, mappings, settings, query etc.  The easier
you make for us to reproduce it, the more likely that somebody will take the
time to look at it.

 1.
 2.
 3.

**Provide logs (if relevant)**:

**Comment by EricDavisX** [comment]
@amitaiporat fyi, transferred to Beats repo - this is where the Agent code lives.

**Comment by threenub** [comment]
+1 on this issue. I had a filebeat-json.log file grow so large it exhausted all the disk space

**Comment by scottdfedorov** [comment]
+1 on this also. The log grew to hundreds of MB in only a few days because of a broken integration causing a LOT of log events. 

We are not setting any value for this config option, we're relying on the default (which I believe should be 10MB). So the issue applies to even installs without changing the default. 

Restarting the agent does appear to rollover the log to another file.

**Comment by andrewkroh** [comment]
Are there any errors in `filebeat-json.log` or `metricbeat-json.log`? My suspicion is that the logger's file rotation is failing because the monitoring-filebeat instance has the file open.

**Comment by greggailly** [comment]
@andrewkroh Our files were filled with errors from failed grok processor. I would tend to agree with your assumption as unchecking "Collect agent logs" fixed the problem, files rotate normally.

**Comment by scottdfedorov** [comment]
There are no error-level events in my filebeat-json log that are related. (The only event is related to a specific input.) There are also no events in the elastic-agent-json.log file.

**Comment by scottdfedorov** [comment]
Disabling the Collect Agent Logs also allowed the logs to rollover for me as well.

**Comment by ruflin** [comment]
@andrewkroh Nice find, was not thinking of this but makes sense. @michalpristas What are the `close_*` options we use today? @kvch What options do you recommend to use for the new filestream input to make sure things work as expected also in Windows?

**Comment by ruflin** [comment]
@jlind23 I think there are two things we need to get fixed here:

* Add different configs to the out own log collection to make sure we close / stop reading files frequently. This should be a quick fix and we should release is as bug fix in 7.16. @kvch should know the right configs to be used
* Be able to configure things globally -> UX fix, later on.

**Comment by kvch** [comment]
@ruflin So you want to have the same configuration on all platforms? I was thinking maybe we could configure rotation based on platform, is it Windows or not. Or is that too much for now?

**Comment by jlind23** [comment]
@kvch @ruflin Shouldn't we then always set the close.on_state_change.removed?

**Comment by ruflin** [comment]
We could have different configs per platform but do we really have to? What do we gain by it? Top priority should be to keep the machine healthy, we rather loose some logs then not rotate on all platforms.

**Comment by kvch** [comment]
We already have a closer configuration for these files. I think we are hitting an issue on Windows that has been reported in other tickets as well. I will find them quickly and reference them here.

**Comment by ruflin** [comment]
@kvch Any update on this / reference to the link? Is this related to filestream or a more generic issue? I think we should have a solution rather soonish.

**Comment by kvch** [comment]
This is probably resolved by https://github.com/elastic/beats/pull/29180
Please reopen if the issue persists.

**Comment by amolnater-qasource** [comment]
Hi @jlind23 
We have revalidated this on **8.1 BC-4** and installed the agent with System and Windows integration.
- We updated the elastic-agent.yml
![2](https://user-images.githubusercontent.com/77374876/156530731-a5239ec7-0521-4a50-a3e8-32767452b417.JPG)

- After approximately 48 we observed below logs:
![1](https://user-images.githubusercontent.com/77374876/156530849-e13713f6-e72a-4185-a9f5-c0e914940227.JPG)

**Build details:**
```
VERSION: 8.1.0-BC4
BUILD: 50428
COMMIT: 015578b81c26a5843747ba53b2fd92d40f0453cb
ARTIFACT LINK: [https://staging.elastic.co/8.1.0-bb72bc41/summary-8.1.0.html](https://staging.elastic.co/8.1.0-bee672a6/summary-8.1.0.html)
```

Further we are running this test on 8.1 BC-5 too.

Could you please help us with any steps for generating more logs.
Please let us know if we are missing anything in this.
cc: @kvch 
Thanks

**Comment by amolnater-qasource** [comment]
Hi @jlind23 
We have revalidated this on **8.1 BC-5** and installed the agent with System and Windows integration.

- After more than 24 hours we observed below logs:
![8](https://user-images.githubusercontent.com/77374876/156745319-632086f3-f781-40ee-9a0c-209e63ea58b0.JPG)

**Logs:**
[logs.zip](https://github.com/elastic/beats/files/8184986/logs.zip)


**Build details:**
```
VERSION: 8.1.0-BC5
BUILD: 50459
COMMIT: 23423b0db7d5ffae1d0578e8d9e2c1afab90cdcf
ARTIFACT: https://staging.elastic.co/8.1.0-d9cd3df6/summary-8.1.0.html
```

cc: @kvch 
Thanks
